builtin/credential/cert/path_login.go

```diff
@@ -409,6 +419,47 @@ func (b *backend) matchesCertificateExtensions(clientCert *x509.Certificate, con
 	return true
 }
 
+// certificateExtensionsMetadata returns the metadata from configured
+// metadata extensions
+func (b *backend) certificateExtensionsMetadata(clientCert *x509.Certificate, config *ParsedCert) map[string]string {
+	// If no metadata extensions are configured, return an empty map
+	if len(config.Entry.MetadataExtensions) == 0 {
+		return map[string]string{}
+	}
+
+	// Build a map of the extensions that should be set as metadata
+	oidMap := map[string]string{}
+	for _, metadataExt := range config.Entry.MetadataExtensions {
+		parts := strings.SplitN(metadataExt, ":", 2)
+		if len(parts) == 1 {
+			// Avoid dots in metadata keys - hard to work with in policy templates.
+			oidMap[metadataExt] = strings.ReplaceAll(metadataExt, ".", "-")
+
+			continue
+		}
+
+		oidMap[parts[0]] = parts[1]
+	}
+
+	metadata := make(map[string]string, len(config.Entry.MetadataExtensions))
+	for _, ext := range clientCert.Extensions {
+		for oidString, alias := range oidMap {
```

Comment:
Do we need to loop through the map in here with the string checking, or could we do something like
`if alias, ok:= oidMap[ext.IdString()]; ok {` ?
The suggestion is less for a time complexity improvement, which I think doesn't matter tbh, and more for readability.